### ISSUE TITLE
Heroic Coven crashing

### ISSUE THREAD

**Issue by kaleide** [BigWigsMods/BigWigs#458]
### What steps will reproduce the problem?  
1.  pull
2.  wait 30 seconds
3.  client crashes

### What version of BigWigs are you using? (Stating 'latest' is not useful)
v90

### Do you have an error log of what happened?
1x BigWigs_Core\BossPrototype.lua:68: table index is nil
BigWigs_Core\BossPrototype.lua:68: in function <BigWigs_Core\BossPrototype.lua:39>
BigWigs_Core\BossPrototype.lua:189: in function <BigWigs_Core\BossPrototype.lua:186>
(tail call): ?
[C]: ?
[string "safecall Dispatcher[1]"]:9: in function <[string "safecall Dispatcher[1]"]:5>
(tail call): ?
...ceAdiBags\libs\AceAddon-3.0\AceAddon-3.0-12.lua:558: in function <...ceAdiBags\libs\AceAddon-3.0\AceAddon-3.0.lua:551>
(tail call): ?
BigWigs_Core\Core.lua:107: in function <BigWigs_Core\Core.lua:105>
BigWigs_Core\Core.lua:118: in function <BigWigs_Core\Core.lua:114>
BigWigs_Core\Core.lua:124: in function <BigWigs_Core\Core.lua:122>
BigWigs_Core\Core.lua:138: in function <BigWigs_Core\Core.lua:133>
BigWigs_Core\Core.lua:168: in function <BigWigs_Core\Core.lua:147>
BigWigs_Core\Core.lua:290: in function `?'
BigWigs_Core\Core.lua:45: in function <BigWigs_Core\Core.lua:40>
[C]: in function `LoadAddOn'
BigWigs\Loader.lua:299: in function <BigWigs\Loader.lua:287>
BigWigs\Loader.lua:309: in function <BigWigs\Loader.lua:306>
BigWigs\Loader.lua:319: in function <BigWigs\Loader.lua:317>
BigWigs\Loader.lua:1165: in function `ZONE_CHANGED_NEW_AREA'
BigWigs\Loader.lua:648: in function `?'
BigWigs\Loader.lua:884: in function <BigWigs\Loader.lua:883>

Locals:
nil

### Any additional information? (example: WoW language if not English) (Attach screenshots here if available, do not link externally)

**Comment by funkydude** [comment]
I don't see what this has to do with you crashing.

1. If WoW crashes, there is no way to get an error from an addon, because it doesn't write addon data.
2. The trace of the error you provided is showing that the module enabled because boss1 existed.

Both of these points tell me that the error you received happened after you logged back into the game from your crash. Whilst the error is valid, it's a side effect of your crash, not a cause.

**Comment by kaleide** [comment]
had 8 people crashing, and it stopped after i disabled the addon

**Comment by funkydude** [comment]
Are you using WeakAuras & have you tried without it?

**Comment by kaleide** [comment]
We ended up leaving it alone. I do run weakauras as well, but only disable BW.
It was only that boss, haven't had a problem the rest of the night.

**Comment by funkydude** [comment]
Since it wasn't your whole raid experiencing it: do you know if the ones that were getting the error were running WeakAuras, and have you shared anything new recently?

**Comment by kaleide** [comment]
I would be willing to bet that most everyone in the raid had WA(guild requirement) and 8 or so of us also have BW. I myself have not accepted anything new.

**Comment by funkydude** [comment]
I just tested LFR and everything worked fine.
Are you saying everyone using BW experienced this crash?

**Comment by funkydude** [comment]
We've now had people test this on mythic and hc and cannot get it to crash, so this must be a conflict with another addon you're using.
Other than WeakAuras, what other addons do you all have in common?

I'd have to recommend trying with only BigWigs enabled, perhaps you can test LFR.

**Comment by Serafim1991** [comment]
The same problem, but not on Coven. When we pulled Garothi  about 8 people are disconnected on some decimations(first time - on 2nd decimation, 2nd time - on fourth decimation). And 2 people have hard lag, but they wasn't disconnected.
Unfortunatelly, my I forget to check BugGrabber and BugSack, so can't say was there some errors about BigWigs.
I can't say that this problem specifically with BigWigs, but when I and other people  disabled it on 3rd pull - no one was disconnected.
Also I tried downgrade BigWigs(to 89) but it doesn't helped.
And the last(maybe it will helped you or other people find the problem)  - today I updated 4 addon: BigWigs, oPie, Raider.IO and LittleWigs. Yesterday there was no such problem.

**Comment by funkydude** [comment]
Thanks for the extra info regarding addons. Unfortunately a disconnect is a whole different situation than a crash and could be caused by any number of things including ISP issues.

BugSack saves all errors, so you can open that sack and look at old errors you previously received.

**Comment by egea-a** [comment]
I've encountered the problem too, and after reconnecting I can't see any bug in BugSack
According to the Twitch client, the only addons recently updated are 

- all  the BigWigs suite (Main addon, Old content and LittleWigs)

- the Raider.io addon

**Comment by funkydude** [comment]
Again, disconnecting is not crashing. For anyone posting here, make sure you clarify which one you are experiencing.

Here are the 2 solutions people need to follow:

1. Disable every single addon expect from the following: **BigWigs**, **BigWigs_Core**, **BigWigs_Plugins**, **BigWigs_Options**, **BigWigs_Antorus**. Then test a raid, which can easily been done in LFR, since no one cares if you play bad there.
2. If the issue still occurs, exit the game. Load up twitch and select BigWigs, hit "delete". When it asks you to also delete the settings, make sure it's ticked, you want this. Now exit Twitch, go into your addons folder and delete every single addon that starts with BigWigs or LittleWigs. Load up Twitch again and install only the main BigWigs addon and try the raid again.

**Comment by egea-a** [comment]
Yeah, sorry about the lack of clarity, what precisely happened was : 
- Really low FPS (about 1-2 FPS)
- Freeze
- "You have been disconnected from the server"

I'll try to do a LFR with no addon later this weekend

**Comment by elvador** [comment]
Please tell us the encounter and what just happend in the fight as well (i.e. Ability x got cast etc.).

**Comment by egea-a** [comment]
It was in mythic, and it happened right around the time she cast the Norganon army ability.
It seemed to happen after a whirling blade too, but I'm not sure about that.

I logged the night here : https://www.warcraftlogs.com/reports/gF1KPM9t7kqLTpBY/
We disabled the addons after the two first wipes. The log itself is missing the end of the fight, as I was the one logging when I got the disconnect.

Hope this helps

**Comment by tyrdolorem** [comment]
This bug is occurring in mythic+ dungeons aswell with the same symptoms several seconds of 1fps followed by a disconnect that requires restarting the game to log back in. First time occurred after the death of the first boss while nothing was happening. Then occurred once on a roaming pack of imps and then every time the second boss spawned imps, we gave it 3 goes before giving up.

I've also had a friend tell me hes had the same problem on the second boss in BRH and the first in antorus.

Mods I updated today, raider.io bigwigs dbm littlewigs Opie and badboy.

**Comment by funkydude** [comment]
Thanks for the info. But at this point we need to figure out if it has anything to do with BigWigs, I've heard from users that haven't updated in 2 weeks that are disconnecting. Please follow the 2 steps I posted above.

**Comment by taccr** [comment]
Happened to my raid today on garothi as soon as apocalypse drive started to cast.
Every single raider with latest bw version dropped to 1-2fps for ~5s and then gets disconnected.
(Looks pretty much like an infinite loop somewhere in the code that causes this)

Everyone who didn't update for a few weeks was not affected.
We then disabled bw for the first boss only. Worked fine everywhere else for us.

I have heard similar reports from friends.

**Comment by funkydude** [comment]
@taccr Have you followed the 2 steps I posted above? Was this the first attempt of the night at the first boss of the night?

**Comment by funkydude** [comment]
Here's two more things you can try:

1. Update to v90.2
2. Delete BigWigs and manually download the old v88

**Comment by Trid3nt1** [comment]
I'd like to add to this - 7 of the 10 members in my raid, the other 3 had DBM, all crashed on pull of Coven. Additionally, when we did M+ CoS and BRH last night, the 3 of the 5 with BigWigs all crashed on the first boss of CoS and 2nd boss of BRH (Sorry, I haven't got their names to hand at the moment)

**Comment by Manta7388** [comment]
Can confirm this issue... got this on Myth Coven with 8 players in raid

**Comment by funkydude** [comment]
Please update to v91 and let us know if the issue is resolved.

**Comment by Trid3nt1** [comment]
I've just done the greater invasion and didn't crash at all - Although the only difference in that scenario is: I wasn't leader, it wasn't in Antorus (if it was an isolated issue to that raid.).

I'm about to go and do M VoTW.. Dunno if that's relevant, but if I crash again I'll let you know.

**Comment by Trid3nt1** [comment]
So for what it's worth, nobody got DC'd and I've done a greater and M now. Unfortunately I won't be raiding again until Monday evening, but I guess others in here will have more answers sooner than that.

This has made me realise how much I rely on BigWigs though, I tried DBM for 5 brief minutes.. never again.

**Comment by Mbrockmo** [comment]
Updated to the newest version amd now me and my group had several disconnects in m+.

It happend in CoS 2nd boss right when the imps spawned. We tried it a few times Sometimes the disconnect happened on the 2nd wave of adds but mostly we got disconnected on the first spawn.

It also happened on the first boss in eye of azshara. Pretty late into the fight we had not  killed the current set of adds and an additional 2 spawned. This made us disconnect. We were not able to reproduce this.

The last dungeon we tried was HoV. We failed on hyria. To wipe us faster we pulled the 3pack before we boss. This also resulted in a disconnect. Tried the same thing on fenryr nothing happened.

I had buggrabber running but it did not catch any errors during this sessions.
